addons/mail/static/src/core/common/core.scss

```diff
@@ -46,12 +46,19 @@
     z-index: 1;
 }
 
-a.o_mail_redirect, a.o_channel_redirect {
-    @include button-variant(rgba($primary, .2), rgba($primary, .2), darken($link-color, 5%), rgba($primary, .3), rgba($primary, .3), darken($link-color, 10%));
+a.o_mail_redirect, a.o_channel_redirect, a.o-discuss-mention {
     @include button-size($btn-padding-y-sm, $btn-padding-x-sm, $btn-font-size-sm, $btn-border-radius-sm);
     padding: map-get($spacers, 1);
 }
 
+a.o_mail_redirect, a.o_channel_redirect {
+    @include button-variant(rgba($primary, .2), rgba($primary, .2), darken($link-color, 5%), rgba($primary, .3), rgba($primary, .3), darken($link-color, 10%));
+}
+
+a.o-discuss-mention {
+    @include button-variant(rgba($primary, .2), rgba($primary, .2), darken($link-color, 5%));
+}
```

Comment:
Not providing `hover` parts will use the default ones, which still show an hover effect.
To cancel hover effects, you can put the same colors as static ones.

Also the mouse hovering shows a clickable cursor. If the mention has no `.o_mail_redirect` nor `.o_channel_redirect`, it should have `cursor: default !important` style